docs/README-emscripten.md

```diff
@@ -1,8 +1,172 @@
+run again when it's time. The `1` parameter does some magic to make
+your main() function end immediately; this is useful because you
+don't want any shutdown code that might be sitting below this code
+to actually run if main() were to continue on, since we're just
+getting started.
+
+There's a lot of little details that are beyond the scope of this
+document, but that's the biggest intial set of hurdles to porting
+your app to the web.
+
+
+## Do you need threads?
+
+If you plan to use threads, they work on all major browsers now. HOWEVER,
+they bring with them a lot of careful considerations. Rendering _must_
+be done on the main thread. This is a general guideline for many
+platforms, but a hard requirement on the web.
+
+Many other things also must happen on the main thread; often times SDL
+and Emscripten make efforts to "proxy" work to the main thread that
+must be there, but you have to be careful (and read more detailed
+documentation than this for the finer points).
+
+Even when using threads, your main thread needs to set an Emscripten
+mainloop that runs quickly and returns, or things will fail to work
+correctly.
+
+You should definitely read [Emscripten's pthreads docs](https://emscripten.org/docs/porting/pthreads.html)
+for all the finer points. Mostly SDL's thread API will work as expected,
+but is built on pthreads, so it shares the same little incompatibilities
+that are documented there, such as where you can use a mutex, and when
+a thread will start running, etc.
+
+
+IMPORTANT: You have to decide to either build something that uses
+threads or something that doesn't; you can't have one build
+that works everywhere. This is an Emscripten (or maybe WebAssembly?
+Or just web browsers in general?) limitation. If you aren't using
+threads, it's easier to not enable them at all, at build time.
+
+If you use threads, you _have to_ run from a web server that has
+[COOP/COEP headers set correctly](https://web.dev/why-coop-coep/)
+or your program will fail to start at all.
+
+If building with threads, `__EMSCRIPTEN_PTHREADS__` will be defined
+for checking with the C preprocessor, so you can build something
+different depending on what sort of build you're compiling.
+
+
+## Audio
+
+Audio works as expected at the API level, but not exactly like other
+platforms.
+
+You'll only see a single default audio device. Audio capture also works;
+if the browser pops up a prompt to ask for permission to access the
+microphone, the SDL_OpenAudioDevice call will succeed and start producing
+silence at a regular interval. Once the user approves the request, real
+audio data will flow. If the user denies it, the app is not informed and
+will just continue to receive silence.
+
+SDL3 allows you to open the audio device at any time; it'll just throw away
+audio data until the user interacts with the page, and then start feeding
+the browser seamlessly, but for SDL2, you need to manage this yourself.

```

Comment:
I'm pretty sure it works automatically in SDL2 as well, at least it did the last time I checked. Also this paragraph seems duplicated below.

Reply:
Random example that definitely uses SDL2 (and a bit old emscripten) https://32blit.github.io/32blit-sdk/examples/shell.html?app=audio-test

Reply:
Oh, yeah, this changed at some point.

Until recently, the SDL device opens, and we use autoResumeAudioContext(), so Emscripten takes care of getting it playing when there was user interaction. However, until that interaction, the audio callback doesn't fire, so for example loopwave.c will start playing the audio from the beginning once you click on the page instead of acting like it was progressing all along.

SDL3 fixed this to use setInterval to call the callback regularly and throw away the data until the device resumes, so things stay in sync.

But also, I forgot I backported those SDL3 improvements to SDL2 in ae7f54f514525cdee3420c03324ddaebb7b3d5f0.

All this to say: yeah, I'll fix the docs.  :)

Reply:
Well, I learned something there. (I should really do a better job of staying up-to-date on this...)